source/Nevermore/Advanced/Queryable/QueryProvider.cs

```diff
@@ -121,31 +103,37 @@ public IAsyncEnumerable<TResult> StreamAsync<TResult>(Expression expression, Can
             return new QueryTranslator(configuration).Translate(expression);
         }
 
-        static async Task<IList> CreateList(IAsyncEnumerable<object> items, Type elementType)
+        IEnumerable ExecuteStream(PreparedCommand command, Type elementType)
         {
-            var listType = typeof(List<>).MakeGenericType(elementType);
-            IList list = (IList)Activator.CreateInstance(listType);
-            await foreach (var item in items.ConfigureAwait(false))
-            {
-                list.Add(item);
-            }
+            return (IEnumerable)GenericStreamMethod.MakeGenericMethod(elementType)
+                .Invoke(queryExecutor, new object[] { command });
+        }
 
-            return list;
+        async Task<IEnumerable> ExecuteStreamAsync(PreparedCommand command, Type elementType, CancellationToken cancellationToken)
+        {
+            var stream = GenericStreamAsyncMethod.MakeGenericMethod(elementType)
+                .Invoke(queryExecutor, new object[] { command, cancellationToken });
+
+            return await AsyncEnumerableAdapter.ConvertToEnumerable(stream, elementType, cancellationToken).ConfigureAwait(false);
         }
 
-        static async ValueTask<T> FirstOrDefaultAsync<T>(IAsyncEnumerable<T> enumerable, CancellationToken cancellationToken)
+        object ExecuteScalar(PreparedCommand command, Type resultType)
         {
-#pragma warning disable CA2007
-            // CA2007 doesn't understand ConfiguredCancelableAsyncEnumerable and incorrectly thinks we need another ConfigureAwait(false) here
-            await using var enumerator = enumerable.ConfigureAwait(false).WithCancellation(cancellationToken).GetAsyncEnumerator();
-#pragma warning restore CA2007
+            return GenericExecuteScalarMethod.MakeGenericMethod(resultType)
+                .Invoke(queryExecutor, new object[] { command });
+        }
 
-            if (await enumerator.MoveNextAsync())
+        static object FirstOrDefault(IEnumerable stream)
```

Comment:
should be `object?` for null-safety

Reply:
Nevermore doesn't have #nullable turned on currently. Another problem for another day 😄